internal/charger/tplink.go

```diff
@@ -150,12 +151,21 @@ func (c *TPLink) execCmd(cmd string) ([]byte, error) {
 	// encode command message
 	// encResult provides the encrypted plug command
 	encCommand := bytes.Buffer{}
-	encCommand.Write([]byte{0, 0, 0, 0}) // BigEndian, unsigned integer msg header
-	var ekey byte = 171                  // Encryption initialization vector
+	var ekey byte = 171 // Encryption initialization vector
 	for i := 0; i < len(cmd); i++ {
 		ekey = ekey ^ cmd[i]
 		encCommand.WriteByte(ekey)
 	}
+	// Pack command msg
+	pkgCommand := bytes.Buffer{}
```

Comment:
Dasist aber wirklich komplex (aber geht offensichtlich). Im standard buffer mit 4x 0 erzeugen und am ende das bigendian in die erhaltenen buf.Bytes() an Stelle 0 rein schreiben.

Reply:
Hier die Funktion https://golang.org/pkg/encoding/binary/#ByteOrder. Damit ersparst Du dir den Buffer hin und her zu kopieren.